### ISSUE TITLE
Multiindex Lightgbm problem

### ISSUE THREAD

**Issue by grayskripko** [alteryx/evalml#1710]
I'm not sure whether the problem arises on evalml side or Lightgbm but I have a problem with multiindex X passing to `AutoMLSearch.search()`

```
Batch 1: (4/9) LightGBM Regressor w/ Imputer            Elapsed:00:01
	Starting cross validation
			Fold 0: Encountered an error.
			Fold 0: All scores will be replaced with nan.
			Fold 0: Please check ...\evalml_debug.log for the current hyperparameters and stack trace.
			Fold 0: Exception during automl search: logical_types contains columns that are not present in dataframe: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16]
			Fold 1: Encountered an error...
```
Sorry, I can't give you a proper example. I still think this is better than nothing.

On my side, I fixed the problem with 
```
X_train.pipe(lambda df: df.set_axis(['_'.join(col).strip() for col in df.columns.values], axis=1))
```

In addition, being still a novice in using the library, but I can't find a proper way to see logs of errors. I see the same information in evalml_debug.log without a clear traceback of the error. This is why I don't know the exact reason for the problem.

**Comment by angela97lin** [comment]
Hi @grayskripko, thank you for filing! RE a proper way to surface the traceback of the error, try:
```
from evalml.automl.callbacks import raise_error_callback
automl = AutoMLSearch(..., error_callback=raise_error_callback)
automl.search()
```

Having that traceback could be helpful for us to understand what is happening and better assist you 😄

**Comment by grayskripko** [comment]
```
Batch 1: (4/14) LightGBM Regressor w/ Imputer            Elapsed:00:01
	Starting cross validation
AutoMLSearch raised a fatal exception: logical_types contains columns that are not present in dataframe: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15]
  File "C:\Users\Gray\AppData\Roaming\Python\Python38\site-packages\evalml\automl\automl_search.py", line 677, in _compute_cv_scores
    cv_pipeline.fit(X_train, y_train)

  File "C:\Users\Gray\AppData\Roaming\Python\Python38\site-packages\evalml\utils\base_meta.py", line 18, in _set_fit
    return_value = method(self, X, y)

  File "C:\Users\Gray\AppData\Roaming\Python\Python38\site-packages\evalml\pipelines\regression_pipeline.py", line 32, in fit
    self._fit(X, y)

  File "C:\Users\Gray\AppData\Roaming\Python\Python38\site-packages\evalml\pipelines\pipeline_base.py", line 197, in _fit
    self._component_graph.fit(X, y)

  File "C:\Users\Gray\AppData\Roaming\Python\Python38\site-packages\evalml\pipelines\component_graph.py", line 89, in fit
    self._compute_features(self.compute_order, X, y, fit=True)

  File "C:\Users\Gray\AppData\Roaming\Python\Python38\site-packages\evalml\pipelines\component_graph.py", line 201, in _compute_features
    component_instance.fit(input_x, input_y)

  File "C:\Users\Gray\AppData\Roaming\Python\Python38\site-packages\evalml\utils\base_meta.py", line 18, in _set_fit
    return_value = method(self, X, y)

  File "C:\Users\Gray\AppData\Roaming\Python\Python38\site-packages\evalml\pipelines\components\estimators\regressors\lightgbm_regressor.py", line 90, in fit
    X_encoded = self._encode_categories(X, fit=True)

  File "C:\Users\Gray\AppData\Roaming\Python\Python38\site-packages\evalml\pipelines\components\estimators\regressors\lightgbm_regressor.py", line 75, in _encode_categories
    X_encoded = _rename_column_names_to_numeric(X_encoded)

  File "C:\Users\Gray\AppData\Roaming\Python\Python38\site-packages\evalml\utils\gen_utils.py", line 235, in _rename_column_names_to_numeric
    return ww.DataTable(X_renamed, logical_types=renamed_logical_types)

  File "C:\Users\Gray\AppData\Roaming\Python\Python38\site-packages\woodwork\datatable.py", line 81, in __init__
    _validate_params(dataframe, name, index, time_index, logical_types,

  File "C:\Users\Gray\AppData\Roaming\Python\Python38\site-packages\woodwork\datatable.py", line 1106, in _validate_params
    _check_logical_types(dataframe, logical_types)

  File "C:\Users\Gray\AppData\Roaming\Python\Python38\site-packages\woodwork\datatable.py", line 1162, in _check_logical_types
    raise LookupError('logical_types contains columns that are not present in '

```

**Comment by dsherry** [comment]
@grayskripko thank you for filing! We're looking into it. We'll follow up if we have more questions for you.

**Comment by gsheni** [comment]
@dsherry I am to reproduce the error
```python
import pandas as pd
import numpy as np
import woodwork as ww
from evalml.utils.gen_utils import _rename_column_names_to_numeric

df = pd.DataFrame([[1,2,3], [10,20,30], [100,200,300]])
df.columns = pd.MultiIndex.from_tuples((("a", "b"), ("a", "c"), ("d", "f")))
df = ww.DataTable(df)
df = _rename_column_names_to_numeric(df)
```

**Comment by angela97lin** [comment]
Thanks @gsheni for the simple repro! Yup, it looks like the issue is that `_rename_column_names_to_numeric` doesn't handle `pd.MultiIndex` very well. In general, I don't think we've done too much testing with passing in dataframes with `MultiIndex` 😬 . I can put up a fix for this particular case and try to run AutoMLSearch to see if anything else breaks.

**Comment by angela97lin** [comment]
@grayskripko We recently released a new version of EvalML (0.18.2) which introduces a fix for MultiIndex issues. Please try it out and let us know if this fixes your original issue! 😁
